File: Anole/HTMLHRElement.cs
using System;
using System.Drawing;
using System.Xml;

namespace Anole
{
	/// <summary>
	/// Summary description for HTMLHRElement.
	/// </summary>
	public class HTMLHRElement : HTMLElement
	{
		public HTMLHRElement(ref HTMLDocument document, HTMLElement parent, XmlNode xnode)
		{
			Initialize(ref document, parent, xnode);
			//TBD: store information attributes about the
			//horizontal rule: line color, thickness, etc.
		}

		public override void Layout(HTMLRenderer renderer, Graphics g)
		{
			HTMLRenderBand band;
			band=renderer.NewBand();
			band.AddItem(new HTMLRenderHRItem(renderer.TargetWidth,this));
			renderer.NewBand();
		}

		
	}
}

File: Anole/HTMLHRElement.vb
Namespace Anole

    Public Class HTMLHRElement : Inherits HTMLElement

        Public Sub New(ByRef document As HTMLDocument, ByVal parent As HTMLElement, ByVal xnode As XmlNode)
            'TBD: store information attributes about the
            'horizontal rule: line color, thickness, etc.
            Initialize(document, parent, xnode)
        End Sub

        Public Overloads Overrides Sub Layout(ByVal renderer As HTMLRenderer, ByVal g As Graphics)
            Dim band As HTMLRenderBand
            band = renderer.NewBand()
            band.AddItem(New HTMLRenderHRItem(renderer.TargetWidth, Me))
            renderer.NewBand()
        End Sub

    End Class

End Namespace
